### ISSUE TITLE
Unable to launch Atom 1.10.0-beta0 on Windows

### ISSUE THREAD

**Issue by bmharsha** [atom/atom#12283]
### Prerequisites

* [x] Can you reproduce the problem in [safe mode](http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode)?
* [x] Are you running the [latest version of Atom](http://flight-manual.atom.io/hacking-atom/sections/debugging/#update-to-the-latest-version)?
* [x] Did you check the [debugging guide](http://flight-manual.atom.io/hacking-atom/sections/debugging/)?
* [X] Did you check the [FAQs on Discuss](https://discuss.atom.io/c/faq)?
* [x] Are you reporting to the [correct repository](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#atom-and-packages)?
* [x] Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3Aatom+-repo%3Aatom%2Felectron) to see if your bug or enhancement is already reported?

### Steps to Reproduce

1. Download Atom ZIp using following [link](https://github.com/atom/atom/releases/download/v1.10.0-beta0/atom-windows.zip)
2. Extract the zip and try launching `atom.exe` , either through GUI or through command prompt 

**Expected behavior:** I was expecting `atom.exe` to launch Atom IDE 

**Actual behavior:** Ended up with following error

```
C:\Users\delete_this_user\Downloads\Atom Beta>atom.exe

C:\Users\delete_this_user\Downloads\Atom Beta>
ENOENT: no such file or directory, lstat 'C:\Users\delete_this_user\.atom'
Error: ENOENT: no such file or directory, lstat 'C:\Users\delete_this_user\.atom
'
    at Error (native)
    at fs.lstatSync (fs.js:887:18)
    at Object.fs.lstatSync (ELECTRON_ASAR.js:228:16)
    at Object.realpathSync (fs.js:1528:21)
    at Object.fs.realpathSync (ELECTRON_ASAR.js:303:29)
    at setupAtomHome (C:\Users\delete_this_user\Downloads\Atom Beta\resources\ap
p.asar\src\main-process\main.js:115:19)
    at start (C:\Users\delete_this_user\Downloads\Atom Beta\resources\app.asar\s
rc\main-process\main.js:26:3)
    at Object.<anonymous> (C:\Users\delete_this_user\Downloads\Atom Beta\resourc
es\app.asar\src\main-process\main.js:263:1)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
```

### Versions
```
>atom --version
Atom    : 1.10.0-beta0
Electron: 0.37.8
Chrome  : 49.0.2623.75
Node    : 5.10.0
```
OS = Windows 2008 R2 server, SP1

### Workaround

Create `.atom` folder manually and execute `atom.exe`

**Comment by Zireael07** [comment]
Did you try with a user without underscores in name?

**Comment by bmharsha** [comment]
> Did you try with a user without underscores in name?

Yes, I did try with a username which didn't had underscore in it, I faced same issue over there.

**Comment by 50Wliu** [comment]
And is there an `.atom` folder present under your user profile (eg `C:\Users\delete_this_user`)?

**Comment by bmharsha** [comment]
> And is there an .atom folder present under your user profile (eg C:\Users\delete_this_user)?

No, I didn't have a `.atom` folder under `C:\Users\delete_this_user` when I encountered this error.

Just a clarification, I created a `.atom` folder manually in following location `C:\Users\delete_this_user\` as a workaround for this issue, I'm able to launch Atom IDE and install plugins after creating this folder manually, this folder was supposed to be created automatically when I launched `atom.exe` for the first time.

**Comment by as-cii** [comment]
Hey @bmharsha, thanks for the report. I think this is a regression caused by #12246, but we have already a fix for it in #12306 that will go out with the next hot-fix to beta. Apologies for this and thanks again for letting us know about this issue! ❤️

**Comment by bmharsha** [comment]
Hello @as-cii , thank you very much for the prompt response!
